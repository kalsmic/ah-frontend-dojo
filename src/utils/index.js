import { shallow } from 'enzyme';
import jwt_decode from 'jwt-decode';

export const baseURL = 'https://ah-backend-dojo-dev.herokuapp.com/api';

export const formatDate = (rawDate) => {
  const longDateTime = new Date(rawDate);

  const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
  const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thur', 'Fri', 'Sat'];

  const month = months[longDateTime.getMonth()];
  const day = days[longDateTime.getDay()];
  const year = longDateTime.getFullYear();
  let date = longDateTime.getDate();

  if (date < 10) {
    date = `0${date}`;
  }

  return `${day} ${date} ${month} ${year}`;
};

export const mockArticle = (
  title = 'This is the first article',
  description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Magna sit amet purus gravida quis blandit turpis. Elit ullamcorper dignissim cras tincidunt lobortis feugiat.',
) => (
  {
    id: 1,
    slug: 'this-is-the-first-article',
    title,
    body: 'An article first',
    description,
    author: 'kalsmic',
    publish_status: false,
    createdAt: '2019-05-17T12:56:21.406849Z',
    updatedAt: '2019-05-07T12:56:21.406908Z',
    delete_status: false,
    tagList: [
      'one',
      'two',
    ],
    time_to_read: 2,
    read_stats: {
      views: 0,
      reads: 0,
    },
    likeCount: [
      {
        likes: 0,
        dislikes: 0,
      },
    ],
  });

const renderComponent = component => shallow(component);

export const shouldContainClass = (component, className) => {
  expect(renderComponent(component).find(className).length).toBe(1);
};

export const shouldContainText = (component, className, text) => {
  expect(renderComponent(component).find(className).text()).toBe(text);
};
/**
*Checks if user session is set
* @param none
* @return {object} {username,email,token} if sessionStorage is set
 * else return user object with empty values
*/
export const isAuthenticated = () => {
  try {
    const token = sessionStorage.getItem('ahToken');

    const UserInfo = jwt_decode(token);

    return {
      username: UserInfo.username,
      email: UserInfo.email,
      token,
    };
  } catch
  (error) {
    // invalid token format
    return {
      username: '',
      email: '',
      token: ''
    };
  }
};
