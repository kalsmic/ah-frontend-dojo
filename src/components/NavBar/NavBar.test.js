// react libraries
import React from 'react';
import { shallow } from 'enzyme';

// components
import { shouldContainClass, shouldContainText } from 'utils';
import { Navbar } from './index';

describe('NavBar component', () => {
  const props = {
    opensignupModal: false,
    openloginModal: false,
    user: {
      username: '',
    },
    logout: jest.fn(),
  };
  const navBar = shallow(<Navbar {...props} />);
  const wrapper = <Navbar {...props} />;


  it('should render without crushing', () => {
    expect(navBar).toMatchSnapshot();
  });

  it('Should contain a navbar class', () => {
    shouldContainClass(wrapper, '.navbar');
  });

  it('Should conatain a branding class', () => {
    shouldContainClass(wrapper, '.navbar__branding');
  });

  it('Should contain a branding name class', () => {
    shouldContainText(wrapper, '.navbar__branding__name', 'Authors\' Haven');
  });

  it('Should contain a moto class', () => {
    shouldContainText(wrapper, '.navbar__branding__moto', 'We provide blogging bliss and areading experience second to none');
  });

  it('Should contain a navigation class', () => {
    shouldContainClass(wrapper, '.navbar__navigation');
  });

  it('Should contain an auth class', () => {
    shouldContainClass(wrapper, '.navbar__navigation__auth');
  });

  it('Should contain an articles class', () => {
    shouldContainClass(wrapper, '.navbar__navigation__articles');
  });

  it('Should contain an articles ul class', () => {
    shouldContainClass(wrapper, '.navbar__navigation__articles__ul');
  });
});
