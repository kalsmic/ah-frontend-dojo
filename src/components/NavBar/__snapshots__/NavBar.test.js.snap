// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render without crushing 1`] = `
<div
  className="navbar"
>
  <div
    className="navbar__branding"
  >
    <div
      className="navbar__branding__name"
    >
      <h1>
        Authors' Haven
      </h1>
    </div>
    <div
      className="navbar__branding__moto"
    >
      <p>
        We provide blogging bliss and a
      </p>
      <p>
        reading experience second to none
      </p>
    </div>
  </div>
  <div
    className="navbar__navigation"
  >
    <div
      className="navbar__navigation__auth"
      id="userInfo"
    >
      <ul>
        <li
          className="navbar__navigation__auth__button"
        >
          <span
            className="navbar__navigation__auth__button--link"
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            |
          </span>
        </li>
        <li
          className="navbar__navigation__auth__button"
        >
          <span
            className="navbar__navigation__auth__button--link"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            tabIndex="0"
          >
             
            Logout
          </span>
        </li>
      </ul>
    </div>
    <div
      className="navbar__navigation__articles"
    >
      <ul
        className="navbar__navigation__articles__ul"
      >
        <li
          className="navbar__navigation__articles__ul__li"
        >
          <Link
            className="navbar__navigation__articles__ul__li--button"
            replace={false}
            to="/"
          >
            Home
          </Link>
        </li>
        <li
          className="navbar__navigation__articles__ul__li"
        >
          <Link
            className="navbar__navigation__articles__ul__li--button"
            replace={false}
            to="/articles"
          >
            Articles
          </Link>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
